stdlib/SparseArrays/src/sparsematrix.jl

```diff
@@ -354,6 +354,58 @@ function copyto!(A::AbstractSparseMatrixCSC, B::AbstractSparseMatrixCSC)
     return A
 end
 
+copyto!(A::AbstractMatrix, B::AbstractSparseMatrixCSC) = _sparse_copyto!(A, B)
+# Ambiguity resolution
+copyto!(A::PermutedDimsArray, B::AbstractSparseMatrixCSC) = _sparse_copyto!(A, B)
+
+function _sparse_copyto!(dest::AbstractMatrix, src::AbstractSparseMatrixCSC)
+    dest === src && return dest
+    z = convert(eltype(dest), zero(eltype(src))) # should throw if not possible
+    isempty(src) && return dest
+    isrc = LinearIndices(src)
+    checkbounds(dest, first(isrc))
+    checkbounds(dest, last(isrc))
+    # If src is not dense, zero out the portion of dest spanned by isrc
+    if length(src) > nnz(src)
+        for i in isrc
+            @inbounds dest[i] = z
+        end
+    end
+    @inbounds for col in 1:size(src, 2), ptr in nzrange(src, col)
```

Comment:
```suggestion
    @inbounds for col in axes(src, 2), ptr in nzrange(src, col)
```

(A hedge against the day when we might have non-standard indexing of `AbstractSparseMatrixCSC`.)